test/unit/src/common/reducers/t_snap-builds.js

```diff
@@ -72,93 +73,83 @@ describe('snapBuilds reducers', () => {
   });
 
   context('FETCH_BUILDS', () => {
-    it('should store fetching status when fetching builds', () => {
-      const action = {
-        type: ActionTypes.FETCH_BUILDS,
-        payload: 'test'
-      };
+    const action = {
+      type: ActionTypes.FETCH_BUILDS,
+      payload: 'test'
+    };
 
+    it('should store fetching status when fetching builds', () => {
       expect(snapBuilds(initialState, action)).toEqual({
         ...initialState,
         isFetching: true
       });
     });
   });
 
-  context('FETCH_BUILDS_SUCCESS', () => {
+  context('FETCH_SNAP_SUCCESS', () => {
+    const SNAP_LINK = 'https://api.launchpad.net/devel/~cjwatson/+snap/godd-test-2';
+
+    const state = {
+      ...initialState,
+      isFetching: true
+    };
+    const action = {
+      type: ActionTypes.FETCH_SNAP_SUCCESS,
+      payload: SNAP_LINK
+    };
 
     it('should stop fetching', () => {
-      const state = {
-        ...initialState,
-        isFetching: true
-      };
+      expect(snapBuilds(state, action).isFetching).toBe(false);
+    });
 
-      const action = {
-        type: ActionTypes.FETCH_BUILDS_SUCCESS,
-        payload: SNAP_ENTRIES
-      };
+    it('should store snap link', () => {
+      expect(snapBuilds(state, action).snapLink).toEqual(SNAP_LINK);
+    });
+  });
+
+  context('FETCH_BUILDS_SUCCESS', () => {
+    const state = {
+      ...initialState,
+      isFetching: true
+    };
 
+    const action = {
+      type: ActionTypes.FETCH_BUILDS_SUCCESS,
+      payload: SNAP_ENTRIES
+    };
+
+    it('should stop fetching', () => {
       expect(snapBuilds(state, action).isFetching).toBe(false);
     });
 
     it('should store builds on fetch success', () => {
-      const state = {
-        ...initialState,
-        isFetching: true
-      };
-
-      const action = {
-        type: ActionTypes.FETCH_BUILDS_SUCCESS,
-        payload: SNAP_ENTRIES
-      };
-
       expect(snapBuilds(state, action).builds).toEqual(SNAP_ENTRIES.map(snapBuildFromAPI));
     });
 
     it('should store success state', () => {
-      const state = {
-        ...initialState,
-        isFetching: true
-      };
-
-      const action = {
-        type: ActionTypes.FETCH_BUILDS_SUCCESS,
-        payload: SNAP_ENTRIES
-      };
-
       expect(snapBuilds(state, action).success).toBe(true);
     });
 
     it('should clean error', () => {
-      const state = {
-        ...initialState,
-        error: 'Previous error'
-      };
-
-      const action = {
-        type: ActionTypes.FETCH_BUILDS_SUCCESS,
-        payload: SNAP_ENTRIES
-      };
-
       expect(snapBuilds(state, action).error).toBe(null);
```

Comment:
The refactoring is incorrect here, rendering the test ineffective.  The initial state for this test must have `error` set.